File: support_tools/swigwin-3.0.12/Examples/test-suite/go/smart_pointer_extend_runme.go
package main

import . "./smart_pointer_extend"

func main() {
	f := NewFoo()
	b := NewBar(f)

	if b.Extension() != f.Extension() {
		panic(0)
	}

	b2 := NewCBase()
	d := NewCDerived()
	p := NewCPtr()

	if b2.Bar() != p.Bar() {
		panic(0)
	}

	if d.Foo() != p.Foo() {
		panic(0)
	}

	if CBaseHello() != p.Hello() {
		panic(0)
	}

	d2 := NewDFoo()

	dp := NewDPtrFoo(d2)

	if DFooSExt(1) != dp.SExt(1) {
		panic(0)
	}

	if d2.Ext(1) != dp.Ext(1) {
		panic(0)
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/smart_pointer_extend_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

smart_pointer_extend

f = Foo();
b = Bar(f);

if (b.extension() != f.extension())
  error
endif


b = CBase();
d = CDerived();
p = CPtr();

if (b.bar() != p.bar())
  error
endif

if (d.foo() != p.foo())
  error
endif

if (b.hello() != p.hello())
  error
endif



d = DFoo();

dp = DPtrFoo(d);

if (d.SExt(1) != dp.SExt(1))
  error
endif

if (d.Ext(1) != dp.Ext(1))
  error
endif

  
